### PR TITLE
chore: clear println in codebase

### PR DESCRIPTION
**Pull request by broccoliSpicy**
issue #3065
println in benchmark code/main function/some test code remains.

### CODE REVIEW

**Review by chebbyChefNEQ — APPROVED** [review]

**Comment by broccoliSpicy** [comment]
https://github.com/lancedb/lance/blob/2377632a47c39211257dca5c05a5830e38e38fcd/rust/lance-index/src/vector/hnsw/builder.rs#L229

@BubbleCal I am not sure how to get rid of this `println`, could you take a look at it?

**Comment by broccoliSpicy** [comment]
> can we apply this lint as well? https://rust-lang.github.io/rust-clippy/master/index.html#print_stdout

looks promising, I think we just need to make it configured to allow `println` in `main`, `benchmark code`, `test code` then we are good to go.

**Comment by wjones127** [comment]
> > can we apply this lint as well? https://rust-lang.github.io/rust-clippy/master/index.html#print_stdout
> 
> looks promising, I think we just need to make it configured to allow `println` in `main`, `benchmark code`, `test code` then we are good to go.

I will add `print_stdout = "deny"` to my PR https://github.com/lancedb/lance/pull/3054

We can make exceptions inline with `#[allow(clippy::print_stdout)]` per file or for specific lines as needed.

**Comment by wjones127** [comment]
> https://github.com/lancedb/lance/blob/2377632a47c39211257dca5c05a5830e38e38fcd/rust/lance-index/src/vector/hnsw/builder.rs#L229
> 
> @BubbleCal I am not sure how to get rid of this `println`, could you take a look at it?

This seems like a good place to use `log::warn!()` instead.

File: integration/duckdb_lance/src/arrow.rs [hunk 1]
```diff
@@ -107,8 +107,7 @@ pub fn to_duckdb_logical_type(data_type: &DataType) -> Result<LogicalType> {
             child.data_type(),
         )?))
     } else {
-        println!("Unsupported data type: {data_type}, please file an issue https://github.com/eto-ai/lance");
-        todo!()
+        todo!("Unsupported data type: {data_type}, please file an issue at https://github.com/lancedb/lance");
```
Comment (BubbleCal) [inline integration/duckdb_lance/src/arrow.rs:110]: `todo!`->`log::warn!`

**Review by BubbleCal — APPROVED** [review]

File: integration/duckdb_lance/src/arrow.rs (hunk 1 above)
Comment (wjones127) [reply integration/duckdb_lance/src/arrow.rs:110]: We are already panicking, so it's reasonable to keep panicking here.

Comment (broccoliSpicy) [reply integration/duckdb_lance/src/arrow.rs:110]: thanks, changed to `panic!`

Comment (broccoliSpicy) [reply integration/duckdb_lance/src/arrow.rs:110]: oh, I mean this line: https://github.com/lancedb/lance/blob/2377632a47c39211257dca5c05a5830e38e38fcd/rust/lance-index/src/vector/hnsw/builder.rs#L229

File: rust/lance-index/src/vector/hnsw/builder.rs [hunk 2]
```diff
@@ -226,7 +226,7 @@ impl HNSW {
         match self.inner.visited_generator_queue.push(visited_generator) {
             Ok(_) => {}
             Err(_) => {
-                println!("visited_generator_queue is full");
+                panic!("visited_generator_queue is full");
```
Comment (wjones127) [inline rust/lance-index/src/vector/hnsw/builder.rs:229]: @BubbleCal @broccoliSpicy Seems like we originally didn't want to panic here. Why is it acceptable to panic here? Should we instead make this a warning?

```suggestion
                log::warn!("visited_generator_queue is full");
```

Comment (broccoliSpicy) [reply rust/lance-index/src/vector/hnsw/builder.rs:229]: yeah, sorry, this is due to a miscommunication between me and @BubbleCal.

**Pull request merged**